mkosi

```diff
@@ -832,6 +833,99 @@ gpgkey={gpg_key}
+#    else:
+#        gpg_key = "https://getfedora.org/static/%s.txt" % FEDORA_KEYS_MAP[args.release]
+
+    if args.mirror:
+        baseurl = "{args.mirror}/distrib/{args.release}/x86_64/media/core/release/".format(args=args)
+        if not check_if_url_exists("%s/media.repo" % baseurl):
+            baseurl = "{args.mirror}/distrib/cauldron/x86_64/media/core/release/".format(args=args)
+
+        release_url = "baseurl=%s" % baseurl
+        updates_url = "baseurl={args.mirror}/distrib/{args.release}/x86_64/media/core/updates/".format(args=args)
+    else:
+        release_url = ("mirrorlist=https://www.mageia.org/mirrorlist/?" +
+                       "release={args.release}&arch=x86_64&section=core&repo=release".format(args=args))
+        updates_url = ("mirrorlist=https://www.mageia.org/mirrorlist/?" +
+                       "release={args.release}&arch=x86_64&section=core&repo=updates".format(args=args))
+
+    with open(os.path.join(workspace, "dnf.conf"), "w") as f:
+        f.write("""\
+[main]
+gpgcheck=1
+
+[mageia]
+name=Mageia {args.release} Core Release
+{release_url}
+gpgkey={gpg_key}
+
+[updates]
+name=Mageia {args.release} Core Updates
+{updates_url}
+gpgkey={gpg_key}
+""".format(args=args,
+           gpg_key=gpg_key,
+           release_url=release_url,
+           updates_url=updates_url))
+    if args.repositories:
+        repos = ["--enablerepo=" + repo for repo in args.repositories]
+    else:
+        repos = ["--enablerepo=mageia", "--enablerepo=updates"]
+
+    root = os.path.join(workspace, "root")
+    cmdline = ["dnf",
+               "-y",
+               "--config=" + os.path.join(workspace, "dnf.conf"),
+               "--best",
+               "--allowerasing",
+               "--releasever=" + args.release,
+               "--installroot=" + root,
+               "--disablerepo=*",
+               *repos,
+               "--setopt=keepcache=1",
+               "--setopt=install_weak_deps=0"]
+
+    # Turn off docs, but not during the development build, as dnf currently has problems with that
+    if not args.with_docs and not run_build_script:
+        cmdline.append("--setopt=tsflags=nodocs")
+
+    cmdline.extend([
+               "install",
+               "systemd",
+               "mageia-release",
+               "passwd"])
+
+    if args.packages is not None:
+        cmdline.extend(args.packages)
+
+    if run_build_script and args.build_packages is not None:
+        cmdline.extend(args.build_packages)
+
+    if args.bootable:
+        cmdline.extend(["kernel", "binutils"])
+
+        # Temporary hack: dracut only adds crypto support to the initrd, if the cryptsetup binary is installed
+        if args.encrypt or args.verity:
+            cmdline.append("cryptsetup")
+
+        if args.output_format == OutputFormat.raw_gpt:
+            cmdline.append("e2fsprogs")
+
+        if args.output_format == OutputFormat.raw_btrfs:
+            cmdline.append("btrfs-progs")
```

Comment:
hmm, any chance we can at least unify the actual dnf command invocation? i.e. add a new function invoke_dnf() or so, which takes the distro-specific stuff as argument, and unifies the stuff shared between fedora + mageia otherwise?